### DISCUSSION TITLE
Detect supported hardware via Bafang Config Tool?

### DISCUSSION THREAD

**Discussion by FoxExe** [danielnilsson9/bbs-fw#257] [General]
I have Bafang BBSHD. Bafang config tool shows this:

> Manufacturer: SZBF
> Model: SN12
> Hardware ver.: 1.0
> Firmware ver.: 5.0.2.1

In wiki i see only hw v1.4 or v1.5 is supported...
But I, nevertheless, flashed the firmware... And it works great!
(STC tool shows "IAP15W4K61S4" as my CPU)
**Maybe should add this information to the wiki?**

**Comment by aalm** [comment]
I think only the STC tool method is in anyway reliable. Someone would have to collect those and make a list,
but even then that information is still coming from the firmware, which anyone can change by flashing.

iirc., adding support for these would bring that info:
https://github.com/danielnilsson9/bbs-fw/blob/1a0c35f2fcdda5dc4c3af6fb47ceb5af9355bc3b/src/firmware/extcom.c#L69-L71

Also, as far as i can tell, there's no revision of BBSHDs generally available this wouldn't work on, so until
a new version appears this is relatively meaningless exercise(trying to check/detect if supported or not).
+given rumours in teh internets, they're even going to simply stop producing these in 2025, so a new hw
version is rather unlikely to appear at this point anymore.

**Comment by dav0000000** [comment]
The STC chip has its own hardware version and firmware. We put our program onto the STC device using it's firmware. We also call our program 'the firmware'. That is where the confusion lies. Very interesting that you have a different STC version though.
